kong/credentials_service.go

```diff
@@ -8,8 +8,8 @@ import (
 	"reflect"
 )
 
-// abstractCredentialService handles credentials in Kong.
-type abstractCredentialService interface {
+// AbstractCredentialService handles credentials in Kong.
+type AbstractCredentialService interface {
```

Comment:
What was the rationale for this and the related changes elsewhere? It looks unrelated to adding the developer role.

I believe the reason that this isn't exposed currently is that it's designed for use within the other credential types in the package, e.g. the BasicAuth service uses it, and that provides the public implementation for adding that specific credential type.

Reply:
I rollback this